remote/client.go

```diff
@@ -178,19 +176,49 @@ func (m *RemoteManager) WatchLeases(ctx context.Context, network string, cursor
 
 	resp, err := m.httpGet(ctx, url)
 	if err != nil {
-		return subnet.WatchResult{}, err
+		return err
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return subnet.WatchResult{}, httpError(resp)
+		return httpError(resp)
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(wr); err != nil {
+		return err
 	}
 
-	wr := subnet.WatchResult{}
-	if err := json.NewDecoder(resp.Body).Decode(&wr); err != nil {
-		return subnet.WatchResult{}, err
+	return nil
+}
+
+func (m *RemoteManager) WatchLeases(ctx context.Context, network string, cursor interface{}) (subnet.LeaseWatchResult, error) {
+	url := m.mkurl(network, "leases")
+
+	wr := subnet.LeaseWatchResult{}
+	err := m.watch(ctx, url, cursor, &wr)
+	if err != nil {
+		return subnet.LeaseWatchResult{}, err
 	}
 	if _, ok := wr.Cursor.(string); !ok {
-		return subnet.WatchResult{}, fmt.Errorf("lease watch returned non-string cursor")
+		return subnet.LeaseWatchResult{}, fmt.Errorf("watch returned non-string cursor")
+	}
+
+	return wr, nil
+}
+
+func (m *RemoteManager) WatchNetworks(ctx context.Context, cursor interface{}) (subnet.NetworkWatchResult, error) {
+	wr := subnet.NetworkWatchResult{}
+	err := m.watch(ctx, m.base+"/", cursor, &wr)
+	if err != nil {
+		return subnet.NetworkWatchResult{}, err
+	}
+
+	// Only return a cursor for the 'watch' operation
+	if cursor != nil {
```

Comment:
Sorry, I think you misunderstood me a bit. Yes, I proposed combining watch+get much like you've done here. However you should always return a cursor. Otherwise the watch won't work -- you kick the watch off by getting all the networks (no cursor passed in) and then watch for increments (using cursor) if you so desire. I think what you have for `WatchLeases` is perfect.